### PR TITLE
Lights fixes

### PR DESCRIPTION
**Pull request by aleksander0m**
The following patches make the android brightness settings work properly.

### CODE REVIEW

**Comment by aleksander0m** [comment]
The first patch isn't generic enough probably, now that I see it, as it fixes the sysfs path to the "pic-backlight" directory...

File: ueventd.rc [hunk 1]
```diff
@@ -13,10 +13,7 @@
 /sys/bus/iio/devices/iio:device* scan_elements/in_anglvel_y_en 0660 system system
 /sys/bus/iio/devices/iio:device* scan_elements/in_anglvel_z_en 0660 system system
 /sys/bus/iio/devices/trigger* 	 name 0660 system system
-/sys/class/backlight*		 brightness 0664 system system
-/sys/class/backlight*		 max_brightness 0664 system system
```
Comment (robherring) [inline ueventd.rc]: It's preferred to set file permissions in ueventd.rc. If the max_brightness line isn't needed, can't we just delete that one line?

My intention is to move all of these to system/core

File: lights/lights.c [hunk 2]
```diff
@@ -159,9 +159,15 @@ static int open_lights(const struct hw_module_t *module, char const *name,
...
 	closedir(dir);
 
+	ALOGI("backlight path: %s\n", backlight_path);
+
 	snprintf(buffer, 1024, "%s/max_brightness", backlight_path);
 	brightness_max = read_int(buffer);
-	ALOGE("JDB: brightness_max: %ld\n", brightness_max);
+
+	if (brightness_max <= 0)
+		ALOGE("invalid max brightness: %ld\n", brightness_max);
```
Comment (robherring) [inline lights/lights.c:168]: Can this really happen with a mainline kernel?

**Review by robherring — CHANGES_REQUESTED** [review]
patches 2 and 4 look good.

File: ueventd.rc (hunk 1 above)
Comment (aleksander0m) [reply ueventd.rc]: Aren't ueventd.rc rules applied *only* for device files exposed via uevents? From some tests with real HW that is what it looked like, and those /sys/class/backlight/*/brightness files were not getting the correct permissions set. In particular, brightness wasn't being set RW.

File: lights/lights.c (hunk 2 above)
Comment (aleksander0m) [reply lights/lights.c:168]: It can if e.g. there's a permission error when trying to read from the brightness file (the reader function returns -errno).

**Comment by robertfoss** [comment]
@aleksander0m Is this PR still in progress?

**Comment by aleksander0m** [comment]
@robertfoss @robherring It is in progress, but I didn't spend much more time on this yet.

The two outstanding issues are the ueventd.rc vs init logic to change permissions, and the 'generic' filename used for brightness control (i.e. not always "pic-backlight").

Regarding the first issue, and as far as I understand, setting permissions on ueventd.rc only applies when there's some device file being notified via uevents; if there is none, as in this case with the lights control, the ueventd.rc permissions don't apply (they will never run), and we need the file permissions changed via init rules. But really I'm not that experienced to say this is always the case, especially since the kernel I used for testing where I validated those assumptions is the upstream Linux kernel and not an android-ized one.

**Comment by aleksander0m** [comment]
@robherring I have now updated this branch, removing the commit that moved the chmod to init.rc, and adding a new one which just removes the ueventd.rc line that changes the max_brightness file permissions. I still left the log updated for when read_int() fails, feel free to leave that out of the merge if you think it really isn't that useful.

**Pull request merged**